src/lib/mqtt/mqtt.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+*
+* Copyright 2018 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Burak Karaboga
+*/
+#include <mosquitto.h>
+
+#include "logMsg/logMsg.h"
+#include "mqtt/mqtt.h"
+
+
+
+/* ****************************************************************************
+*
+* Globals
+*/
+struct mosquitto *mosq;
+const char*      mqttHostname;
+unsigned short   mqttPortNumber;
+unsigned short   mqttKeepAlivePeriod;
+
+
+
+/* ****************************************************************************
+*
+* mqttOnPublishCallback -
+* 
+*/
+void mqttOnPublishCallback(struct mosquitto *mosq, void *userdata, int mid)
+{
+  //FIXME: Do we have a usecase for the callback? To be resolved during second PR
+  LM_W(("PoC: in the publish callback"));
```

Comment:
When the callback is invoked? When the notification has been sucessfully published in MQTT broker?

Reply:
Exactly, it is invoked when a message is successfully sent to the MQTT broker.

Currently I see no use for the callback function other than hosting a proper log message. I'll move the log message below  from `sendMqttNotification` to here.

    LM_I(("MQTT notification sent to %s:%d", mqttHostname, mqttPortNumber));

Reply:
Yes, logs seem to be the only purpose (and very important when debugging, by the way :)

However, I'd suggest not moving the message about MQTT notification sents, but have two ones. In particular, for `sendMqttNotification` we could use the same we have now and for `mqttOnPublishCallback` we can use:

```
LM_I(("MQTT notification sucessfully published ...", ...);
```

including extra information (if any) about that (in the place where `...` are).

Reply:
Fixed in 4896535